File: agora_plugin/android/src/main/java/com/agora/agora_plugin/AgoraRendererView.java
package com.agora.agora_plugin;

import android.view.SurfaceView;
import android.view.View;

import io.flutter.plugin.platform.PlatformView;

public class AgoraRendererView implements PlatformView {
    private SurfaceView mSurfaceView;
    private final long uid;

    AgoraRendererView(SurfaceView surfaceView, int uid) {
        this.mSurfaceView = surfaceView;
        this.uid = uid;
    }

    @Override
    public View getView() {
        return mSurfaceView;
    }

    @Override
    public void dispose() {
        mSurfaceView = null;
    }
}

File: agora_plugin/android/src/main/kotlin/com/example/agora_plugin/AgoraRendererView.kt
package com.example.agora_plugin

import android.view.SurfaceView
import android.view.View
import io.flutter.plugin.platform.PlatformView

class AgoraRendererView internal constructor(private var mSurfaceView: SurfaceView?, uid: Int) : PlatformView {
    private val uid: Long
    override fun getView(): View? {
        return mSurfaceView
    }

    override fun dispose() {
        mSurfaceView = null
    }

    init {
        this.uid = uid.toLong()
    }
}